ggshield/sca/output/schemas.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass, field
+from typing import List, Optional, Type, cast
+
+import marshmallow_dataclass
+from pygitguardian.models import Base, BaseSchema, FromDictMixin
+
+
+@dataclass
+class SCAVulnerability(Base, FromDictMixin):
+    severity: str
+    summary: str
+    cve_ids: List[str] = field(default_factory=list)
+    created_at: Optional[str] = None
+    fixed_version: Optional[str] = None
+
+
+SCAVulnerabilitySchema = cast(
+    Type[BaseSchema], marshmallow_dataclass.class_schema(SCAVulnerability, BaseSchema)
+)
+
+SCAVulnerability.SCHEMA = SCAVulnerabilitySchema()
+
+
+@dataclass
+class SCAVulnerablePackageVersion(Base, FromDictMixin):
+    full_name: str
+    version: str
+    ecosystem: str
+    dependency_type: str
+    vulns: List[SCAVulnerability] = field(default_factory=list)
+
+
+SCAVulnerablePackageVersionSchema = cast(
+    Type[BaseSchema],
+    marshmallow_dataclass.class_schema(SCAVulnerablePackageVersion, BaseSchema),
+)
+
+SCAVulnerablePackageVersion.SCHEMA = SCAVulnerablePackageVersionSchema()
+
+
+@dataclass
+class SCALocationVulnerability(Base, FromDictMixin):
+    filepath: str
+    package_vulns: List[SCAVulnerablePackageVersion] = field(default_factory=list)
```

Comment:
```suggestion
    package_vulns: List[SCAVulnerablePackageVersion] = []
```

Can't we just do this?

Reply:
No, that's not the same, see https://docs.python.org/3/library/dataclasses.html#default-factory-functions.